### PR TITLE
blog: Add Authenticate with OpenID Connect and Apache APISIX post

### PR DESCRIPTION
**Pull request by nfrankel**
Screenshot 2023-03-02 at 15 28 49

As usual, please get the images from my own site after Sunday, store them on apiseven.com and change the links to the images before merging

### CODE REVIEW

**Review by juzhiyuan — COMMENTED** [review]
Preview Link: https://deploy-preview-1518--apache-apisix.netlify.app/blog/2023/03/09/authenticate-openid-connect/

Please register an account at https://drive.api7.ai/ and upload images, this PR has many image links to replace :)

**Comment by juzhiyuan** [comment]
Hi @guoqqqi, please review this and make sure the steps are ok to follow :)

File: blog/en/blog/2023/03/09/authenticate-openid-connect.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+          secret: ${{SESSION_SECRET}}                                          #6
+#END
+```
+
+1. Catch-all route to the underlying web app
+2. Plugin configuration parameters. Values depend on the exact provider (see below)
+3. OpenID Connect can use a Discovery endpoint to get all necessary OAuth endpoints. See [OpenID Connect Discovery 1.0 spec](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig) for more information
+4. Where to redirect when the authentication is successful. It mustn't clash with any of the explicitly defined routes. The plugin creates a dedicated route there to work its magic.
+5. Default scope
+6. Key to encrypt session data. Put whatever you want.
```
Comment (guoqqqi) [inline blog/en/blog/2023/03/09/authenticate-openid-connect.md:124]: When the secret is too short, the following error log will appear. Do we need to specify here that the key length is at least 16?
```
err:failed to check the configuration of plugin openid-connect err: property "session" validation failed: property "secret" validation failed: string too short, expected at least 16, got 4 
```

**Review by guoqqqi — COMMENTED** [review]
For other LGTM, There are just pictures that haven't been updated yet

**Comment by yzeng25** [comment]
@guoqqqi @juzhiyuan I just added a commit to update all 7 images in the pull request, it should be ready to go now.

**Review by yzeng25 — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Comment by yzeng25** [comment]
Just checked all images are correct in blog summary page and blog content page. 

link: https://apisix.apache.org/blog/2023/03/09/authenticate-openid-connect/

截屏2023-06-14 19 49 03

**Pull request merged**